### PR TITLE
Refs [Task][30572] Handle event for Training History Screen

### PR DESCRIPTION
**Pull request by NguyenTaTrung**
## Related Tickets
- [#Task_30572: Handle event for Training History Screen](https://edu-redmine.sun-asterisk.vn/issues/30572)

## WHAT
- Handle event for Training History Screen

## HOW
- Bind data: get data pour into recyclerview
- Item recyclerview click show a list of exercises you have exercised
- Fix some bug

## Review Checklist

Category | View Point | Description | Expected Reviewer Answer | Self review | Reviewer2 (name)
--- | --- | --- | --- | --- | ---
Conventions | Does the code follow Sun* coding style and coding conventions? | https://github.com/framgia/coding-standards/tree/master/eng/android | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Redmine | Does the ticket follow Sun* Redmine working process?  | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md| YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |  | YES |<li>- [x] yes</li>|<li>- [ ] yes</li>  
## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/streetworkout/ui/history/TrainingHistoryViewModel.kt [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    init {
+        repository.getHistoryTraining()
+            .subscribeOn(io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                _trainingHistory.value = it
+            }, {
+                _error.value = it.message.toString()
+            }).addTo(disposables)
```
Comment (datdt-1935) [inline app/src/main/java/com/example/streetworkout/ui/history/TrainingHistoryViewModel.kt]:
```suggestion
            })
            .addTo(disposables)
```

File: app/src/main/java/com/example/streetworkout/ui/home/HomeFragment.kt [hunk 2]
```diff
@@ -35,21 +41,32 @@ class HomeFragment : BaseFragment() {
...
             challengeAdapter.submitList(it.challenge)
             beginAdapter.submitList(it.beginner)
             normalAdapter.submitList(it.normal)
             advanceAdapter.submitList(it.advance)
         })
 
-        homeViewModel.error.observe(viewLifecycleOwner, Observer {
+        historyInformation.observe(viewLifecycleOwner, Observer {
+            textViewTotalTraining.text = it[0].toString()
+            textViewTotalKcal.text = it[1].toString()
```
Comment (datdt-1935) [inline app/src/main/java/com/example/streetworkout/ui/home/HomeFragment.kt]: nếu `List` rỗng hoặc có 1 phần tử thì crash à em?

File: app/src/main/java/com/example/streetworkout/ui/home/HomeViewModel.kt [hunk 3]
```diff
@@ -26,4 +35,22 @@ class HomeViewModel(private val categoryRepository: CategoryRepository) : RxView
             })
             .addTo(disposables)
     }
+
+    fun getTrainingHistoryInformation() {
+        Observable.zip(
```
Comment (datdt-1935) [inline app/src/main/java/com/example/streetworkout/ui/home/HomeViewModel.kt]: dùng `combineLatest` đi em

File: app/src/main/java/com/example/streetworkout/ui/home/HomeViewModel.kt [hunk 4]
```diff
@@ -26,4 +35,22 @@ class HomeViewModel(private val categoryRepository: CategoryRepository) : RxView
...
+            trainingHistoryRepository.getHistoryTraining().toObservable(),
+            trainingHistoryRepository.getTotalKcalTraining().toObservable(),
+            trainingHistoryRepository.getTotalTimeTraining().toObservable(),
+            Function3<List<Training>, Int, Int, List<Int>> { training, totalKcal, totalTime ->
```
Comment (datdt-1935) [inline app/src/main/java/com/example/streetworkout/ui/home/HomeViewModel.kt]: trả về `Triple(Int, Int, Int)` chứ không nên dùng `List` như này

File: app/src/main/java/com/example/streetworkout/ui/home/HomeViewModel.kt [hunk 5]
```diff
@@ -26,4 +35,22 @@ class HomeViewModel(private val categoryRepository: CategoryRepository) : RxView
...
+                mutableListOf(training.size, totalKcal, totalTime)
+            }
+        )
+            .subscribeOn(io())
+            .observeOn(AndroidSchedulers.mainThread())
+            .subscribe({
+                _historyInformation.value = it
+            }, {
+                _error.value = it.message.toString()
+            }).addTo(disposables)
```
Comment (datdt-1935) [inline app/src/main/java/com/example/streetworkout/ui/home/HomeViewModel.kt]:
```suggestion
            })
            .addTo(disposables)
```

File: app/src/main/java/com/example/streetworkout/ui/training/TrainingViewModel.kt [hunk 6]
```diff
@@ -99,6 +96,7 @@ class TrainingViewModel(
     }
 
     fun finishTraining(level: Double) {
+        Log.d("TAG", "$level")
```
Comment (datdt-1935) [inline app/src/main/java/com/example/streetworkout/ui/training/TrainingViewModel.kt]: bỏ log

File: app/src/main/java/com/example/streetworkout/ui/training/TrainingViewModel.kt [hunk 7]
```diff
@@ -127,7 +125,7 @@ class TrainingViewModel(
 
     private fun calculatorKcal(level: Double) {
         val recipe =
-            timeTraining.div(60).times(3.5).times(level).times(weightUser)
+            timeTraining.toDouble().div(60).times(3.5).times(level).times(weightUser)
```
Comment (datdt-1935) [inline app/src/main/java/com/example/streetworkout/ui/training/TrainingViewModel.kt]: những số này nên để ra constant

File: app/src/main/java/com/example/streetworkout/ui/home/HomeFragment.kt (hunk 2 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/example/streetworkout/ui/home/HomeFragment.kt]: E quên check null :D

File: app/src/main/java/com/example/streetworkout/ui/home/HomeViewModel.kt (hunk 3 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/example/streetworkout/ui/home/HomeViewModel.kt]: Dạ vâng a

File: app/src/main/java/com/example/streetworkout/ui/training/TrainingViewModel.kt (hunk 6 above)
Comment (NguyenTaTrung) [reply app/src/main/java/com/example/streetworkout/ui/training/TrainingViewModel.kt]: Úi e quên mất

**Pull request merged**